### ISSUE TITLE
Future of the admin-project

### ISSUE THREAD

**Issue by mkrauser** [muesli/beehive-admin#1]
Do you have any concrete plans for the future of this project? 

I just wanted to ask, because I would like to start working on https://github.com/muesli/beehive/issues/243 and I may require some additions to the admin-gui.

Do you want to stick with ember or are there plans to migrate to react/angular/vue? 
Also the dependencies are pretty outdated and not locked (which was not possible at the time without extensions).

Maybe you already have ideas about that. Thank you!

**Comment by muesli** [comment]
Yeah, I'm working on refactoring the entire admin interface. I'll stick with Ember since it's what I know best and currently just have too little time to try out new JS frameworks.

Hope to get this finished up real soon - I could probably push it to a branch if you want to help out.

![image](https://user-images.githubusercontent.com/146378/57743206-67e15900-76c4-11e9-875b-7b9541eec156.png)

**Comment by mkrauser** [comment]
Wow, that looks really great! 

I can certainly try to help you out there. But I have to familiarize myself with ember first.

**Comment by jonasgeiler** [comment]
Hello Guys!

Last week I've started to redesign Beehive myself out of boredom.
Not sure how far the progress is with the new admin site shown above by @muesli, but I just thought I could use the practice and you guys could use a beautiful interface.

Not only did I redesign the whole page, I've also migrated to Svelte.js as it is super fast and lightweight and I'm very familiar with it. Hope this is not a problem! I know that you are most familiar with Ember.js, but in my opinion it's a bit "overkill" for such a project. (Especially the `npm install` takes a really long time...)
The old admin page loaded about 1,10MB of javascript code. My new site only about 66KB (not finished tho but it'll stay below 100KB).

Also I'm using TailwindCSS for all the CSS stuff.

ANYWAYS
Here are some WIP screenshots:

![image](https://user-images.githubusercontent.com/10259118/99924957-72aa8e80-2d3c-11eb-9279-1be9c9395ec9.png)
---
![image](https://user-images.githubusercontent.com/10259118/99924962-763e1580-2d3c-11eb-91d2-72a06c0dcc32.png)
---
![image](https://user-images.githubusercontent.com/10259118/99925106-1136ef80-2d3d-11eb-8979-56e8ce426bee.png)
---
![image](https://user-images.githubusercontent.com/10259118/99924986-8eae3000-2d3c-11eb-880a-f6f80d77b1f1.png)
---
![image](https://user-images.githubusercontent.com/10259118/99924976-82c26e00-2d3c-11eb-89b5-aac6883a9c97.png)
---
![image](https://user-images.githubusercontent.com/10259118/99925024-b00f1c00-2d3c-11eb-9266-577424a51dd4.png)

Definitely let me know what you think!
I've also done some UX improvements, not just UI, not sure if you like them...

You can see progress on my fork: [Skayo/beehive-admin](https://github.com/Skayo/beehive-admin)

**Comment by pbek** [comment]
@Skayo, it really looks great! Good timing, I was about to start something with VueJs. 😅 
Creating and editing chains doesn't work yet, correct?

**Comment by jonasgeiler** [comment]
> @Skayo, it really looks great! Good timing, I was about to start something with VueJs. 😅 
> Creating and editing chains doesn't work yet, correct?

Yup I haven't started working on that.

Glad you like it! 😊

**Comment by pbek** [comment]
> Yup I'm haven't started working on that.

Awesome! I might chime in a bit if we know what we need. 😁 
... provided @muesli also likes the code base.

**Comment by pbek** [comment]
Chains are looking great too, @Skayo!

<img width="1235" alt="image" src="https://user-images.githubusercontent.com/1798101/100536435-d6dbbf80-3220-11eb-9ab8-7404ca3c72d7.png">

**Comment by corrantho** [comment]
Hello, 

I see above a lot of nice screenshots for the improvement of the admin UI. 
I also would like to contribute to this project but it seems in a blurry state. 
In order to focus our efforts in the same direction, could we have a guidance on which branch to work? 
Should we consider the code provided by Skayo has the base of the new version? If yes, maybe could we have a "v2" branch in this repo so we can also contribute.

beehive is a really great tool and improving the UI (especially the ergonomy) could help new users to adopt it.

Thanks a lot for reading my comment.

**Comment by muesli** [comment]
I fully agree, Beehive needs a lot of help in the UI/admin interface department and @Skayo's efforts look very promising. My only worry here is that I won't be able to help out here as I have literally no experience with Svelte.js. Ideally we have a dedicated team of people who really sign up for this job, so we can delegate and split the work between the frontend and backend of Beehive. Any help and contributions are welcome as this is indeed Beehive's big showstopper right now.

**Comment by corrantho** [comment]
I understand your worry about the language to use and I think it should be a topic to discuss. In my opinion, to maximize the chances to find contributors, the framework that will be used should be stable with a large community. 
I also think it should be compatible with the below criteria:
- easy to learn/adopt, 
- cross platform (desktop, mobile, ...)
- extensible
- well documented

For example, https://ionicframework.com/ is corresponding to the above criteria. 
The UI is based on web components, we make our own if the ones provided are not fitting with our needs and it's compatible with Angular, React, or Vue.

@muesli What would you think about the below next action point list?
- discuss and select a framework
- specify a list of requirements for the V2

@pbek @Skayo would you be interested in working on the V2? What is your opinion on this?

On my side, I'm willing to learn any framework if it's not too exotic.

**Comment by pbek** [comment]
I'm currently mostly only into Vue.js and could now and then drop by for some contributions. 😁
